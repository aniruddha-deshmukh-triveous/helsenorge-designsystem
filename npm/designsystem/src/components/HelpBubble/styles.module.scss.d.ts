export type Styles = {
  anchorlink: string;
  anchorlink__icon: string;
  helpbubble: string;
  'helpbubble__child-wrapper': string;
  'helpbubble__close-wrapper': string;
  'helpbubble__link-button': string;
  'helpbubble__link-button__icon': string;
  'helpbubble--initial-render': string;
  'helpbubble-arrow': string;
  'helpbubble-arrow--initial-render': string;
  'helpbubble-arrow--over': string;
  'helpbubble-arrow--under': string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
