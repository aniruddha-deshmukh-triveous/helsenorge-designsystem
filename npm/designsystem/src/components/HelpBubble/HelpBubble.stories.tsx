import React, { useRef } from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import HelpBubble, { HelpBubbleVariant } from './HelpBubble';
import Icon from '../Icons';
import HelpSign from '../Icons/HelpSign';

export default {
  title: 'Components/HelpBubble',
  component: HelpBubble,
  argTypes: {
    children: {
      control: 'text',
      defaultValue:
        'Dette er en HelpBubble. Aliquip aute consectetur eiusmod nisi ullamco aliquip adipisicing cupidatat reprehenderit nulla in Lorem sint.',
    },
    showBubble: {
      control: 'boolean',
      defaultValue: true,
    },
    variant: {
      control: 'select',
      options: HelpBubbleVariant,
      defaultValue: HelpBubbleVariant.positionautomatic,
    },
  },
} as ComponentMeta<typeof HelpBubble>;

export const Default: ComponentStory<typeof HelpBubble> = (args: any) => {
  const controllerRef = useRef<SVGSVGElement>(null);

  return (
    <div>
      <span>
        {
          'Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris.'
        }
      </span>
      <div style={{ position: 'relative', display: 'inline' }}>
        <Icon ref={controllerRef} svgIcon={HelpSign} />
        <HelpBubble {...args} onLinkClick={action('Mer hjelp clicked')} onClose={action('Bubble closed')} controllerRef={controllerRef}>
          {args.children}
        </HelpBubble>
      </div>
      <span>
        {
          'Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris.'
        }
      </span>
    </div>
  );
};

export const OnText: ComponentStory<typeof HelpBubble> = (args: any) => {
  const controllerRef = useRef<HTMLSpanElement>(null);

  return (
    <div>
      <span>
        {
          'Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris.'
        }
      </span>
      <div style={{ position: 'relative', display: 'inline' }}>
        <span ref={controllerRef} style={{ display: 'inline-block', color: 'red' }}>
          {'Jeg er en tooltip tekst.'}
        </span>
        <HelpBubble {...args} onClose={action('Bubble closed')} controllerRef={controllerRef}>
          {args.children}
        </HelpBubble>
      </div>
      <span>
        {
          'Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris. Dolore quis tempor culpa exercitation laboris. Enim enim pariatur veniam nisi ad quis fugiat magna occaecat nisi. Excepteur ad ad duis quis mollit ex. Laboris duis velit eu eiusmod ea in ex aliqua. Est proident proident dolor veniam nostrud aliquip occaecat dolore sunt laboris nisi ipsum Lorem nisi. Magna sunt ex occaecat amet id. Eiusmod magna Lorem proident est occaecat pariatur sunt esse nostrud. Mollit reprehenderit velit veniam amet ipsum veniam et. Dolor sunt adipisicing enim dolore quis qui occaecat esse quis ut fugiat laboris.'
        }
      </span>
    </div>
  );
};
