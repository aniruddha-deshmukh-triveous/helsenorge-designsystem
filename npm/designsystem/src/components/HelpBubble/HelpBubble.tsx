import classNames from 'classnames';
import React from 'react';

import { AnalyticsId } from '../../constants';
import AnchorLink from '../AnchorLink';
import Close from '../Close';

import styles from './styles.module.scss';

interface HelpBubbleVariant {
  automatic: 'automatic';
  triggertop: 'triggertop';
  triggerbottom: 'triggerbottom';
}

interface HelpBubbleProps {
  /** Name to display in the avatar. Will be truncated to the first two characters. */
  children: string;
  /** Adds custom classes to the element. */
  className?: string;
  /** Determines the placement of the helpbubble */
  variant?: HelpBubbleVariant;
  /** Determines the placement of the bubble arrow */
  triggerTop?: boolean;
  /** Visible text on the link */
  linkText?: string;
  /** Url the link leads to */
  linkUrl?: string;
  /** Sets the data-testid attribute. */
  testId?: string;
}

const HelpBubble: React.FC<HelpBubbleProps> = props => {
  const { children, className = '', triggerTop = false, linkText, linkUrl, testId } = props;

  const helpBubbleClasses = classNames(styles.helpbubble, className);
  const triangleClasses = classNames(styles['helpbubble__arrow'], {
    [styles['helpbubble__arrow--under']]: !triggerTop,
    [styles['helpbubble__arrow--over']]: triggerTop,
  });
  return (
    <div className={helpBubbleClasses} data-testid={testId} data-analyticsid={AnalyticsId.Avatar}>
      <div className={styles['helpbubble__child-wrapper']}>
        {children}
        {linkUrl && <AnchorLink href={linkUrl}>{linkText && linkText !== undefined ? linkText : 'Mer hjelp'}</AnchorLink>}
      </div>
      <div className={styles['helpbubble__close-wrapper']}>
        <Close small />
      </div>
      <div className={triangleClasses} />
    </div>
  );
};

export default HelpBubble;
