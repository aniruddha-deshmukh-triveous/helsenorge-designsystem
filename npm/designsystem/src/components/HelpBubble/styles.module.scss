@import '../../scss/_spacers.scss';
@import '../../scss/_breakpoints.scss';
@import '../../scss/_palette.scss';
@import '../../scss/_font-settings.scss';
@import '../AnchorLink/styles.module.scss';

.helpbubble {
  display: flex;
  position: fixed;
  align-items: flex-start !important;
  max-width: 373px;
  text-align: start;
  font-family: inherit;
  font-size: $font-size-sm;
  line-height: $lineheight-size-sm;
  background-color: $white;
  z-index: 3;
  visibility: visible;

  border: getSpacer(4xs) solid $plum600;
  border-radius: 9px;
  box-shadow: 0 2px 18px 0 rgba(0, 0, 0, 0.15);

  &--initial-render {
    visibility: hidden;
  }

  @media (min-width: map-get($grid-breakpoints, lg)) {
    font-size: $font-size-md;
    line-height: $lineheight-size-md;
  }

  &__child-wrapper {
    display: flex;
    width: max-content;
    flex-direction: column;
    align-items: flex-start;
    overflow-wrap: anywhere;
    row-gap: getSpacer(2xs);
    padding: getSpacer(s) 0 getSpacer(m) getSpacer(s);

    &--no-close-button {
      padding: getSpacer(s) getSpacer(s) getSpacer(m);
    }

    > p {
      margin: 0;
    }
  }

  &__link-button {
    all: unset;
    @include anchorlink;
  }

  &__close-wrapper {
    padding: getSpacer(2xs) getSpacer(2xs) 0px 0px;
  }
}

.helpbubble-arrow {
  width: 0;
  height: 0;
  position: fixed;
  border-style: solid;
  border-color: transparent;
  z-index: 4;

  &--over {
    border-width: 10px 10px;
    border-bottom-color: $white;
    -webkit-filter: drop-shadow($plum600 0px 2px 6px 2px);
    filter: drop-shadow(0px -0.185rem 0px $plum600);
  }
  &--under {
    border-width: 10px 10px;
    border-top-color: $white;
    -webkit-filter: drop-shadow($plum600 0px 2px 6px 2px);
    filter: drop-shadow(0px 0.2rem 0px $plum600);
  }

  &--initial-render {
    visibility: hidden;
  }
}
