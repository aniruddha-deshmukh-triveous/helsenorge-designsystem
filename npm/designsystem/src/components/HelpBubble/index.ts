import HelpBubble from './HelpBubble';
export * from './HelpBubble';
export default HelpBubble;
