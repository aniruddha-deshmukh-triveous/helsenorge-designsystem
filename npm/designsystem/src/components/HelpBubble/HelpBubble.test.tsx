import React from 'react';
import { screen, render } from '@testing-library/react';
import HelpBubble, { HelpBubbleVariant } from './HelpBubble';
import userEvent from '@testing-library/user-event';

describe('Gitt at HelpBubble skal vises', (): void => {
  describe('Når den skal vises vanlig', (): void => {
    it('Så vises HelpBubble som vanlig', (): void => {
      const { container } = render(
        <HelpBubble showBubble testId="test01">
          {'Test tekst'}
        </HelpBubble>
      );

      const bubble = screen.getByTestId('test01');
      const child = screen.getByText('Test tekst');
      const closeButton = screen.getByRole('button');

      expect(bubble).toBeInTheDocument();
      expect(child).toBeInTheDocument();
      expect(closeButton).toBeInTheDocument();

      expect(bubble).toHaveClass('helpbubble helpbubble--below');
      expect(child).toHaveClass('helpbubble__child-wrapper');
      expect(closeButton).toHaveClass('close close--small');

      expect(container).toMatchSnapshot();
    });
  });

  describe('Når variant er positionbelow', (): void => {
    it('Så vises HelpBubble riktig', (): void => {
      render(
        <HelpBubble variant={HelpBubbleVariant.positionbelow} showBubble testId="test01">
          {'Test tekst'}
        </HelpBubble>
      );

      const bubble = screen.getByTestId('test01');

      expect(bubble).toHaveClass('helpbubble helpbubble--below');
    });
  });

  describe('Når variant er positionabove', (): void => {
    it('Så vises HelpBubble riktig', (): void => {
      render(
        <HelpBubble variant={HelpBubbleVariant.positionabove} showBubble testId="test01">
          {'Test tekst'}
        </HelpBubble>
      );

      const bubble = screen.getByTestId('test01');

      expect(bubble).toHaveClass('helpbubble helpbubble--above');
    });
  });

  describe('Når onLinkClick er satt', (): void => {
    it('Så vises ekstra knapp med onLinkClick callback riktig', (): void => {
      const onLinkClick = jest.fn();
      render(
        <HelpBubble showBubble linkText={'Egen link tekst'} onLinkClick={onLinkClick} testId="test01">
          {'Test tekst'}
        </HelpBubble>
      );

      const allButtons = screen.getAllByRole('button');
      const link = screen.getByText('Egen link tekst');
      userEvent.click(link);

      expect(allButtons.length).toBe(2);
      expect(link).toBeInTheDocument();
      expect(link).toHaveClass('helpbubble__link-button');
      expect(onLinkClick).toHaveBeenCalledTimes(1);
    });
  });

  describe('Når linkUrl er satt', (): void => {
    it('Så vises ekstra knapp med linkUrl riktig', (): void => {
      render(
        <HelpBubble showBubble linkText={'Egen link tekst'} linkUrl={'/'} testId="test01">
          {'Test tekst'}
        </HelpBubble>
      );

      const link = screen.getByText('Egen link tekst');
      userEvent.click(link);

      expect(link).toBeInTheDocument();
      expect(link).toHaveClass('anchorlink');
      expect(link).toHaveAttribute('href', '/');
    });
  });

  describe('Når onLinkClick og linkUrl ikke er satt', (): void => {
    it('Så vises ikke ekstra knapp for link', (): void => {
      render(
        <HelpBubble showBubble testId="test01">
          {'Test tekst'}
        </HelpBubble>
      );

      const allButtons = screen.getAllByRole('button');

      expect(allButtons.length).toBe(1);
    });
  });

  describe('Når onClose er satt', (): void => {
    it('Så kalles onClose callback riktig', (): void => {
      const onClose = jest.fn();
      render(
        <HelpBubble onClose={onClose} showBubble testId="test01">
          {'Test tekst'}
        </HelpBubble>
      );

      const close = screen.getByRole('button');
      userEvent.click(close);

      expect(onClose).toHaveBeenCalledTimes(1);
    });
  });
});

describe('Gitt at HelpBubble ikke skal vises', (): void => {
  describe('Når den rendres', (): void => {
    it('Så vises HelpBubble ikke', (): void => {
      render(<HelpBubble testId="test01">{'Test tekst'}</HelpBubble>);

      const bubble = screen.queryByTestId('test01');
      const child = screen.queryByText('Test tekst');

      expect(bubble).not.toBeInTheDocument();
      expect(child).not.toBeInTheDocument();
    });
  });
});
