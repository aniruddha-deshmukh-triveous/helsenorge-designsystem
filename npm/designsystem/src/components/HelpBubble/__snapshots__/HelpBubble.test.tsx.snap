// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gitt at HelpBubble skal vises Når den skal vises vanlig Så vises HelpBubble som vanlig 1`] = `
<div>
  <div
    class="helpbubble helpbubble--above"
    data-analyticsid="help-bubble"
    data-testid="test01"
    style="left: 0px; width: 0px; transform: translateY(0px);"
  >
    <div
      class="helpbubble__child-wrapper"
    >
      Test tekst
    </div>
    <div
      class="helpbubble__close-wrapper"
    >
      <button
        aria-label="Lukk"
        class="close close--small"
        data-analyticsid="close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="hnds-style-icon "
          data-analyticsid="icon"
          fill="#126F87"
          height="38"
          role="img"
          style="min-width: 38px; min-height: 38px;"
          viewBox="0 0 48 48"
          width="38"
        >
          <path
            d="M25.773 24l10.432-10.432-1.772-1.773-10.432 10.433-10.433-10.433-1.773 1.772L22.229 24 11.795 34.433l1.773 1.772 10.433-10.433 10.432 10.433 1.772-1.773z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    class="helpbubble-arrow helpbubble-arrow--under"
    style="transform: translate(0px, 0px);"
  />
</div>
`;
